chapters/data-processing.tex

```diff
@@ -0,0 +1,725 @@
+contains almost exactly same information as the raw data,
+but in a format that is easier to interact with.  
+
+\end{fullwidth}
+
+%------------------------------------------------
+
+
+\section{Identifying the identifier(s)}
+
+% Intro
+The very first step in creating analysis-friendly datasets
+is understanding the data acquired,
+and translating this understanding into an intuitive format.
+This section discusses what steps may be needed to make sure that each row
+in a dataset represents one observation.
+This may seem trivial, but should not be taken for granted.
+Getting to such a format may be harder than expected,
+and the \textbf{unit of observation}\sidenote{\textbf{Unit of observation:}
+	the unit described by the data. In datasets, it is ideally what each row represents. \url{
+		https://dimewiki.worldbank.org/wiki/Unit_of_Observation}}
+ may be ambiguous in many raw datasets.
+In this session, we will discuss the ideal format to handle data,
+which we call a \textit{tidy} format.
+
+
+\subsection{Unique identifiers}
+
+% Uniquely and fully identifying variable
+The first step in tidying data is to understand its unit of observation
+(what makes a row),
+and find out which variable or set of variables is the \textbf{unique identifier}\sidenote{\url{
+		https://dimewiki.worldbank.org/ID\_Variable\_Properties}} for each observation in your dataset.
+Ensuring that observations are uniquely and fully identified
+is arguably the most important step in data cleaning.
+It may be the case that the variables expected to be unique identifiers are either incomplete or contain duplicates.
+It is also possible for a dataset to have no unique identifier, 
+or that the identifier involves a long string that is difficult to work with, such as a full name.
+In such cases, cleaning begins by carefully creating a variable that uniquely identifies the data.
+Note that while modern survey tools create unique identifiers for each data submission,
+that is not the same as having a unique ID variable for each observation in the sample,
+as there can be multiple submissions for the same observation.
+The unique identifier will be used to link data on observations in one dataset 
+with data on the same observations in other datasets, 
+such as other rounds of data collection and different data sources.
+Therefore, you must keep a canonical dataset of all observations in your data.
+This \textbf{master dataset}\sidenote{\url{
+		https://dimewiki.worldbank.org/Master\_Data\_Set}}
+is discussed in detail in Chapter 3, 
+and contains a record of all units ever encountered for each relevant level of observation,
+as well as their corresponding unique identifiers.
+
+
+% Merchand
+Checking for duplicated entries and making sure that each observation has a unique identifier are usually part of data quality monitoring.
+The lack of a unique identifier should be addressed as soon as data is received,
+so you can keep track of observations.
```

Comment:
```suggestion
```
We are making the case that the first step is to check for unique identifier, but in the first sentence I highlighted here, we say that it is a part of _data quality monitoring_ which is the section after this section on tidy data. That is strange.

And the second data says that this should be addressed as soon as data is received, but you have already said above that the first cleaning task is to do exactly that.

I think both these sentences can be deleted.